### ISSUE TITLE
guard.denies is always true

### ISSUE THREAD

**Issue by wuzi** [RomainLanz/adonis-guard#4]
After following the [instructions](https://github.com/RomainLanz/adonis-guard/blob/master/instructions.md) I made a gate but it is always denying, am I missing something?

```js
// start/acl.js
'use strict'

const Gate = use('Gate')

Gate.define('canSee', (user, resource) => {
    return true
})
```

```js
// Controller
'use strict'

class TestController {
    async index({ response, guard }) {
        if (guard.denies('canSee')) {
            return response.status(403).json({ message: 'denied' })
        }
        return response.json({ message: 'not denied'  })
    }
}

module.exports = TestController
```

**Comment by RomainLanz** [comment]
Hey @Wuzi! 👋 

Are you authenticating your routes?
By default `guard.denies()` will be true if you aren't authenticated.

```js
// It will use per default the authenticated user or return false if not authenticated
guard.allows('gateName/Policy Method', resource)
// It will use per default the authenticated user or return true if not authenticated 
guard.denies('gateName/Policy Method', resource)
guard.allows('gateName/Policy Method', resource, user)
guard.denies('gateName/Policy Method', resource, user)
```

**Comment by wuzi** [comment]
Hey!

Yes, I am using jwt method and passing the `authorization` on header.

The route has `.middleware('auth')`.

I'm using the api-only template, I didn't have the `'Adonis/Middleware/AuthInit'` middleware, so I added both.

If I remove `AuthInit` it has the same result.

**Comment by RomainLanz** [comment]
Could you please verify the value of `auth.user` in your controller?

**Comment by wuzi** [comment]
```js
class TestController {
    async index({ response, guard, auth }) {
        if (guard.denies('canSee')) {
            return response.status(403).json({ message: auth.user })
        }
        return response.json({ message: 'not denied'  })
    }
}
```
It returns my user object

```json
{
	"message": {
		"id": 1,
		"name": "Test User",
		"email": "testy@email.com",
		"password": "$2a$10$tEFne2/FWokMpBGWmyCDc.blzbq0GTNdLXkmw77EeAFwhK4sBY.hq",
		"phone": null,
		"role": "admin",
		"created_at": "2018-09-28 13:10:49",
		"updated_at": "2018-09-28 13:10:49"
	}
}
```

**Comment by RomainLanz** [comment]
And how have you registered the middleware?

**Comment by wuzi** [comment]
```js
// kernel.js
'use strict'

/** @type {import('@adonisjs/framework/src/Server')} */
const Server = use('Server')

const globalMiddleware = [
  'Adonis/Middleware/BodyParser',
  'Adonis/Middleware/GuardInit'// tried with and without AuthInit
]

const namedMiddleware = {
  auth: 'Adonis/Middleware/Auth',
  can: 'Adonis/Middleware/Can'
}

const serverMiddleware = [
  'Adonis/Middleware/Static',
  'Adonis/Middleware/Cors'
]

Server
  .registerGlobal(globalMiddleware)
  .registerNamed(namedMiddleware)
  .use(serverMiddleware)
```

```js
// routes.js
'use strict'

/** @type {import('@adonisjs/framework/src/Route/Manager'} */
const Route = use('Route')

Route.get('/', 'TestController.index').middleware('auth')
```

**Comment by RomainLanz** [comment]
Okay, I get the issue.

The `GuardInit` middleware use `auth.user` to authenticate the user and since `globalMiddleware` runs before `namedMiddleware` the variable is still not set.

Let me see how I can fix this. 👍

**Comment by RomainLanz** [comment]
@Wuzi May you try to add `AuthInit` middleware before `GuardInit` and set up your default authentication scheme to `jwt`?

**Comment by wuzi** [comment]
Sure, to set default authentication scheme, its just setting `authenticator: 'jwt'` in `config/auth.js`?
If so, it was already set, I also added AuthInit, but got same result.

**Comment by thetutlage** [comment]
This commit https://github.com/adonisjs/adonis-auth/commit/3de0a82e8bd82086b68bb32225253bef368f6a28 should fix the issue. Try it and lemme know

**Comment by wuzi** [comment]
It's working now! 👍 
Thank you both!

**Comment by thetutlage** [comment]
Nice. Released `@adonisjs/auth@3.0.7`

**Comment by RomainLanz** [comment]
👍
